### PR TITLE
Add support for rectangular (box-styled) halo

### PR DESCRIPTION
**Pull request by stephanr**
This PR is a (smal) reworked part of the larger PR #1204

It adds the possibility to use rectangular helo effects when rendering text.
This is similar to the way deegree2 rendered halo as boxes.

It also contains an edge-case fix so that when halos are set (<> 0) it will be rendered with at least a size of one pixel.

**EDIT:**
For the test to work, it was necessary to include and load a font so that the test was run with the same font.
The Fonts distributed with each JDK distribution differ and can as well differ on the platforms (win/mac/linux).

**EDIT2:**
The previously added Font was replaced with an maven dependency.
The Font is licensed, under "SIL OPEN FONT LICENSE Version 1.1" see also https://search.maven.org/artifact/com.mpobjects.jasperreports.font/jasperreports-fonts-liberation/2.1.2/jar

### CODE REVIEW

**Comment by stephanr** [comment]
Fixed merge conflict by rebase.

File: deegree-services/deegree-webservices-handbook/src/main/asciidoc/renderstyles.adoc [hunk 1]
```diff
@@ -874,7 +874,7 @@ The first parameter is the hatching angle, the second is the line spacing in the
 </PolygonSymbolizer>
 ----
 TIP: For of the shelf hates, which will create nice results, use the mark symbol `shape://slash`, `shape://backslash`
-or `shape://times` for  45°, `shape://horline` for 0° and `shape://vertline` for 90° hatches.
+or `shape://times` for  45�, `shape://horline` for 0� and `shape://vertline` for 90� hatches.
```
Comment (tfr42) [inline deegree-services/deegree-webservices-handbook/src/main/asciidoc/renderstyles.adoc:877]: Please review the encoding of this file. The special character for deegree is not displayed correctly.

**Review by tfr42 — CHANGES_REQUESTED** [review]
Please check if the  TTF files may be added via a maven dependency.

File: deegree-services/deegree-webservices-handbook/src/main/asciidoc/renderstyles.adoc (hunk 1 above)
Comment (stephanr) [reply deegree-services/deegree-webservices-handbook/src/main/asciidoc/renderstyles.adoc]: The encoding was fixed in https://github.com/deegree/deegree3/pull/1407/commits/ec252f76e0908acc3bc94ceb3cc28419c74c2376

**Review by tfr42 — APPROVED** [review]
